### PR TITLE
CGImport metadata fixes

### PR DESCRIPTION
**Pull request by ondrejbudai**
This PR brings in some fixes for our Koji integration - more specifically, the CGImport metadata. All changes were proposed by the maintainers of Brew.

Here's a snapshot of metadata osbuild-composer sends with this patch applied:

```json
{
  "metadata_version": 0,
  "build": {
    "build_id": 1,
    "task_id": 1,
    "name": "name",
    "version": "version",
    "release": "release",
    "source": "",
    "start_time": 1603702833,
    "end_time": 1603703024,
    "extra": {
      "typeinfo": {
        "image": {}
      }
    }
  },
  "buildroots": [
    {
      "id": 1,
      "host": {
        "os": "Fedora release 32 (Thirty Two)",
        "arch": "x86_64"
      },
      "content_generator": {
        "name": "osbuild",
        "version": "1"
      },
      "container": {
        "type": "none",
        "arch": "x86_64"
      },
      "tools": [],
      "components": [
        {
          "type": "rpm",
          "name": "libgcc",
          "version": "10.0.1",
          "release": "0.11.fc32",
          "arch": "x86_64",
          "sigmd5": "84fc907a5047aeebaf8da1642925a417",
          "signature": null
        }
      ],
      "output": [
        {
          "buildroot_id": 1,
          "filename": "name-version-release.x86_64.qcow2",
          "filesize": 514392064,
          "arch": "x86_64",
          "checksum_type": "md5",
          "checksum": "84a28613e512005b9b31271dcc485beb",
          "type": "image",
          "components": [
            {
              "type": "rpm",
              "name": "python3-pyserial",
              "version": "3.4",
              "release": "7.fc32",
              "arch": "noarch",
              "sigmd5": "378cb32f9f850b275ac4e04d21e8144b",
              "signature": null
            }
          ],
          "extra": {
            "image": {
              "arch": "noarch"
            }
          }
        }
      ]
    }
  ]
}
```

### CODE REVIEW

**Comment by ondrejbudai** [comment]
Pushed a new version that also adds the signatures to RPMs.

**Review by teg — CHANGES_REQUESTED** [review]
Very nice! Thanks for going through all this. Only very minor nitpicks. We don't yet know the osbuild version, should we leave that at `0` with a TODO for now?

File: internal/distro/distro.go [hunk 1]
```diff
@@ -193,6 +194,20 @@ func GetHostDistroName() (string, bool, error) {
 	return name, beta, nil
 }
 
+// GetRedHatRelease returns the content of /etc/redhat-release
+// without the trailing new-line.
+func GetRedHatRelease() (string, error) {
+	raw, err := ioutil.ReadFile("/etc/redhat-release")
```
[the remark this answers is not in the document]
Comment (ondrejbudai) [reply internal/distro/distro.go:200]: From Fedora:

```
$cat /etc/os-release
NAME=Fedora
VERSION="33 (Workstation Edition)"
ID=fedora
VERSION_ID=33
VERSION_CODENAME=""
PLATFORM_ID="platform:f33"
PRETTY_NAME="Fedora 33 (Workstation Edition)"
ANSI_COLOR="0;38;2;60;110;180"
LOGO=fedora-logo-icon
CPE_NAME="cpe:/o:fedoraproject:fedora:33"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f33/system-administrators-guide/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=33
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=33
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
VARIANT="Workstation Edition"
VARIANT_ID=workstation
$ cat /etc/redhat-release
Fedora release 33 (Thirty Three)
```

From RHEL:
```
$ cat /etc/os-release 
NAME="Red Hat Enterprise Linux"
VERSION="8.4 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="8.4"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Red Hat Enterprise Linux 8.4 Beta (Ootpa)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:8.4:beta"
HOME_URL="https://www.redhat.com/"
DOCUMENTATION_URL="https://access.redhat.com/documentation/red_hat_enterprise_linux/8/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 8"
REDHAT_BUGZILLA_PRODUCT_VERSION=8.4
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.4 Beta"
$ cat /etc/redhat-release 
Red Hat Enterprise Linux release 8.4 Beta (Ootpa)
```

As you can see, on RHEL, `redhat-release` == `PRETTY_NAME` in `os-release`. However, this isn't true on Fedora. So I just went with what they suggested.

File: internal/kojiapi/server.go [hunk 2]
```diff
@@ -131,6 +134,17 @@ func (h *apiHandlers) PostCompose(ctx echo.Context) error {
...
 		imageRequests[i].arch = arch.Name()
 		imageRequests[i].filename = imageType.Filename()
+
+		filename := fmt.Sprintf(
+			"%s-%s-%s.%s%s",
+			request.Name,
+			request.Version,
+			request.Release,
+			ir.Architecture,
+			splitExtension(imageType.Filename()),
```
[the remark this answers is not in the document]
Comment (ondrejbudai) [reply internal/kojiapi/server.go:144]: I thought of this solution too and I don't have a strong opinion...

**Comment by ondrejbudai** [comment]
> Very nice! Thanks for going through all this. Only very minor nitpicks. We don't yet know the osbuild version, should we leave that at `0` with a TODO for now?

Sure, will do it.

File: internal/kojiapi/server.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (teg) [reply internal/kojiapi/server.go:144]: Maybe leave that as a future improvement, let's land this first.

**Comment by ondrejbudai** [comment]
I force-pushed a new version. All commits are now giving credits to Lubomír and Tomáš from the Brew team. The osbuild version was changed o `"0"` with a TODO added. I don't see any action items so I think this is ready to be merged.

File: internal/distro/distro.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (teg) [reply internal/distro/distro.go:200]: Let's revisit this in a future PR (if at all), I'm not a fan of relying on the non-standard `redhat-release`, but not clear what to do instead.

**Review by teg — APPROVED** [review]

**Pull request merged**